07_safety/juvenile-arrests/juvenile-arrests-place-all.Rmd

```diff
@@ -0,0 +1,876 @@
+    
+Recode race and ethnicity for Group B observations.
+```{r}
+# Group B codes for race and ethnicity do not match those used for Group A
+arrests %>%
+  group_by(group, race_of_arrestee, ethnicity_of_arrestee) %>%
+  count()
+
+# No arrests for Group B are classified as violent or property crimes
+arrests %>%
+  group_by(group, property, violent) %>%
+  count()
+
+
+# Making my best guess to match Group B race/ethnicity codes to Group A codes
+
+## Recode race
+arrests <- arrests %>%
+  mutate(
+    race_of_arrestee = if_else(race_of_arrestee=="w", "white", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="b", "black", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="i", "american indian/alaskan native", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="a", "asian", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="p", "native hawaiian or other pacific islander", race_of_arrestee),
+    race_of_arrestee = if_else(race_of_arrestee=="u", "unknown", race_of_arrestee)
+  )
+
+## Recode ethnicity
+arrests <- arrests %>%
+  mutate(
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="h", "hispanic origin", ethnicity_of_arrestee),
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="n", "not of hispanic origin", ethnicity_of_arrestee),
+    ethnicity_of_arrestee = if_else(ethnicity_of_arrestee=="u", "unknown", ethnicity_of_arrestee)
+  )
+
+```
+
+58,052 observations have `unknown` ethnicity and 75,253 have missing ethnicity
+```{r}
+arrests %>% 
+  group_by(race_of_arrestee, ethnicity_of_arrestee) %>% 
+  count()
+
+arrests %>% 
+  group_by(ethnicity_of_arrestee) %>% 
+  count()
+
+```
+
+
+### 3.4 Aggregate arrest counts to the reporting agency level
+Aggregate arrests by agency and finish race construction
+
+Note that this metric includes the following subgroups by race, sex, and age subgroups, so those subgroups are also aggregated in this section:
+    * Race: white, Black, Hispanic, Asian/other
+    * Sex: male, female
+    * Age: ages 10-14, ages 15-17
+```{r}
+arrests_agency <- arrests %>%
+  filter(age >= 10) %>%
+  group_by(year, ori) %>%
+  summarize(
+    arr_total = n(),
+    # arr_prop = sum(property==1),
+    # arr_viol = sum(violent==1),
+    # Race subgroups
+    race_aian = sum(race_of_arrestee == "american indian/alaskan native"),
+    race_asian = sum(race_of_arrestee == "asian"),
+    race_black = sum(race_of_arrestee == "black"),
+    race_nhpi = sum(race_of_arrestee == "native hawaiian or other pacific islander"),
+    race_unkno = sum(race_of_arrestee == "unknown"),
+    race_white = sum(race_of_arrestee == "white"),
+    ethn_hispanic = sum(ethnicity_of_arrestee == "hispanic origin"),
+    ethn_nonhisp = sum(ethnicity_of_arrestee == "not of hispanic origin"),
+    # Sex subgroups
+    sex_male = sum(sex_of_arrestee == "male"),
+    sex_female = sum(sex_of_arrestee == "female"),
```

Comment:
this should be `sex_female = sum(sex_of_arrestee == "f|female")`

Reply:
Thanks for catching this. I re-coded values of `f` to be `female` for `sex_of_arrestee` earlier in the file, so this no longer needs changed here.